config/configopaque/opaque.go

```diff
@@ -0,0 +1,33 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configopaque // import "go.opentelemetry.io/collector/config/configopaque"
+
+import (
+	"bytes"
+	"encoding"
+)
+
+// String alias that is marshaled in an opaque way.
+type String string
+
+var _ encoding.TextMarshaler = String("")
+
+// MarshalText marshals into a string of '*' of length equal to the opaque string.
+func (s String) MarshalText() ([]byte, error) {
+	return bytes.Repeat([]byte("*"), len(s)), nil
+}
+
+// Headers is a string map with opaque values for configuration fields that set HTTP headers.
+type Headers map[string]String
```

Comment:
Do we need this? I think can easily be defined where needed without any "duplicate code"

Reply:
I felt like it was a common enough pattern that we could add it, but it's not strictly necessary, no. If you don't feel the same let's remove it, we can always add it back later